core/user/user.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package user
+
+import (
+	"fmt"
+	"time"
+)
+
+type User struct {
+	CreatedAt   time.Time
+	DisplayName string
+	Name        string
+}
+
+// UserPassword type attempts to protect it against printouts
+// because we get the password as a plain text over the wire
+type UserPassword struct {
+	password string
+}
+
+func NewUserPassword(password string) UserPassword {
+	return UserPassword{
+		password: password,
+	}
+}
+
+func (up UserPassword) String() string {
+	return ""
+}
+
+// Format implements the Formatter interface from fmt
```

Comment:
@cderici, do you think for both this func and the one below that we should put a quick comment in the func description saying "Stops the plain text password from being outputted"?

Reply:
ignoring based on convo w/ @tlm